fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/Lambda.java

```diff
@@ -0,0 +1,162 @@
+ * After bind, x -> x : arguments("x") -> children: Expression(x) ArrayItemReference(x)
+ */
+public class Lambda extends Expression {
+    final List<String> argumentNames;
+
+    /**
+     * constructor
+     */
+    public Lambda(List<String> argumentNames, Expression lambdaFunction) {
+        super(lambdaFunction);
+        this.argumentNames = argumentNames;
+    }
+
+    public Lambda(List<String> argumentNames, Expression lambdaFunction, List<ArrayItemReference> arguments) {
+        super(ImmutableList.<Expression>builder().add(lambdaFunction).addAll(arguments).build());
+        this.argumentNames = argumentNames;
+    }
+
+    public Lambda(List<String> argumentNames, List<Expression> children) {
+        super(children);
+        this.argumentNames = argumentNames;
+    }
+
+    /**
+     * make slot according array expression
+     * @param arrays array expression
+     * @return item slots of array expression
+     */
+    public ImmutableList<ArrayItemReference> makeArguments(List<Expression> arrays) {
+        Builder<ArrayItemReference> builder = new ImmutableList.Builder<>();
+        for (int i = 0; i < arrays.size(); i++) {
+            Expression array = arrays.get(i);
+            String name = argumentNames.get(i);
+            Preconditions.checkArgument(array.getDataType() instanceof ArrayType, "lambda must receive array");
+            builder.add(new ArrayItemReference(name, array));
+        }
+        return builder.build();
+    }
+
+    public String getLambdaArgumentName(int i) {
+        return argumentNames.get(i);
+    }
+
+    public ArrayItemReference getLambdaArgument(int i) {
+        return (ArrayItemReference) children.get(i + 1);
+    }
+
+    public List<ArrayItemReference> getLambdaArguments() {
+        return children.stream()
+                .skip(1)
+                .map(e -> (ArrayItemReference) e)
+                .collect(Collectors.toList());
+    }
+
+    public List<String> getLambdaArgumentNames() {
+        return argumentNames;
+    }
+
+    public Expression getLambdaFunction() {
+        return child(0);
+    }
+
+    @Override
+    public <R, C> R accept(ExpressionVisitor<R, C> visitor, C context) {
+        return visitor.visitLambda(this, context);
+    }
+
+    public Lambda withLambdaFunctionArguments(Expression lambdaFunction, List<ArrayItemReference> arguments) {
+        return new Lambda(argumentNames, lambdaFunction, arguments);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Lambda that = (Lambda) o;
+        return argumentNames.equals(that.argumentNames)
+                && Objects.equals(children(), that.children());
+    }
+
+    @Override
+    public String toSql() {
+        StringBuilder builder = new StringBuilder();
+        builder.append(String.format("%s -> %s", argumentNames,
```

Comment:
argumentNames is List, List toString use '[]' instead of '()' as container. so i think u should use string join to generate right sql string

Reply:
done